climat/reader.py

```diff
@@ -1,11 +1,11 @@
 from pathlib import Path
-from netCDF4 import Dataset
+import xarray as xr
 from typing import Dict
 
 
-def get_all_data(dirpath: Path) -> Dict[str, Dataset]:
-    dfs = {}
-    for afile in dirpath.glob("*.nc"):
-        print(f"Reading file {afile}...")
-        dfs[afile.stem] = Dataset(afile)
-    return dfs
+def read_data(dirpath: Path, pattern: str) -> Dict[str, xr.Dataset]:
```

Comment:
We might have other file types. I saw that some files are csv, so we might want to add a condition on the type for the reading. This can be done using if afilsuffix == ".grib" ...